### PR TITLE
Refactor site menu hide/show animation

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Refactor animation for main header for a better effect.

[Untitled.mov](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/OgXegTwxM9IeuXHZyw5h/6f30e74f-6233-4dce-bd7d-c31c6a5a89f6/Untitled.mov)

Remove duplication of header content.

## Justify why they are needed

After discussions with Zak and Wille.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1566** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1566" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1566?utm_source=stack-comment).

File: apps/store/src/components/Header/useScrollState.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      const isAbove = latest <= threshold
+      const isScrollingDown = scrollY.getPrevious() - latest < 0
+
+      if (isAbove) {
+        return setScrollState((prev) => {
+          return prev === 'SCROLL_UP' ? 'SCROLL_UP' : 'ABOVE'
+        })
+      }
+
+      return setScrollState((prev) => {
```
Comment (alebedev) [inline apps/store/src/components/Header/useScrollState.ts:25]: Would this be easier to read?
```
if (isScrollingDown) {
  if (prev === 'ABOVE' || prev === 'BELOW') {
    return 'BELOW'
  } else {
    return 'SCROLL_DOWN'
  }
} else {
  return 'SCROLL_UP'
}
```

No inverted condition, no ternary

File: apps/store/src/components/Header/Header.tsx [hunk 2]
```diff
@@ -1,92 +1,102 @@
...
   return (
-    <>
-      <Wrapper opaque={opaque} overlay={overlay}>
-        {headerContent}
+    <GhostWrapper style={{ position: defaultPosition, backgroundColor }}>
+      <Wrapper
+        initial={initial}
+        variants={animationVariants}
+        animate={animate}
+        transition={{ ease: EASE_IN_OUT_CUBIC }}
```
Comment (alebedev) [inline apps/store/src/components/Header/Header.tsx:59]: Can you profile it and check that this does not re-render excessively due too new object passed in every time?

**Review by alebedev — APPROVED** [review]
Looks good!  Some questions inside

File: apps/store/src/components/Header/Header.tsx [hunk 3]
```diff
@@ -1,13 +1,37 @@
...
+    y: '-150%',
+  },
+  HIDE: {
+    y: '-150%',
+    transition: { duration: 0 },
+  },
+} as const
+
+// Source: https://easings.co · easeInOutCubic
+const TRANSITION: Transition = { ease: [0.65, 0.05, 0.36, 1] }
```
Comment (gustaveen) [inline apps/store/src/components/Header/Header.tsx:32]: Have we talked with Zak and Wille on what easings we want to have? Thinking we can make it part of the theme and have a few easing for different use cases

**Review by gustaveen — APPROVED** [review]
Nice improvement! 🎉

**Comment by robinandeer** [comment]
Right, yes the ease function was suggested by Zak -- but let's show them and see what they think! @gustaveen

**Comment by KarnellSchultz** [comment]
@robinandeer Nice stuff 💅 ! Looks really really slick now

**Pull request merged**